### DISCUSSION TITLE
Is it possible to update a saved model

### DISCUSSION THREAD

**Discussion by arunkumar790** [alan-turing-institute/sktime#6899] [Q&A] [answered]
Hi,

I am implementing time series forecasting for manufacturing, after performance evaluation, I am saving the fitted model into a directory using "mlflow_sktime.save_model(sktime_model=model, path=file_path)".

I am using the saved model for future prediction "fut_pred = mlflow_sktime.load_model(model_uri=file_path).predict(fh=fh)"
Is it possible to update the saved model with the latest data? Is it required to rerun the fitting with the latest data and recreate a new file? Do we have options like update_predict_single or update which are used to update the fitted forecaster?

I tried finding a solution with my search but couldn't get one. Thanks for your support.

**Comment by fkiraly** [comment, marked as answer]
Loading and running `update` should work - does that not work for you? If not, this might be a bug.

I thought what would work: 

```python
loaded_f = mlflow_sktime.load_model(model_uri=file_path)
loaded_f.update(new_data)
mlflow_sktime.save_model(sktime_model=model, path=file_path)
```

This is a load-update-save cycle.

**Reply by arunkumar790** [reply]
Thanks a lot @fkiraly it was an amazing help. I was trying to update the loaded model but that didn't work seems I was following the wrong cycle, after following your steps it worked. My apologies

                    load_f = mlflow_sktime.load_model(model_uri=file_path)
                    pred = load_f.predict(fh=fh)
                    pred_inter = load_f.predict_interval(fh=fh, coverage=coverage)
                    load_f.update(pred)
                    shutil.rmtree(file_path)
                    mlflow_sktime.save_model(sktime_model=load_f, path=file_path)

**Reply by fkiraly** [reply]
no worries!
